controllers/apps/components/consensus/consensus.go

```diff
@@ -286,10 +287,33 @@ func (r *ConsensusSet) HandleRoleChange(ctx context.Context, obj client.Object)
 	return nil, nil
 }
 
-func (r *ConsensusSet) HandleHA(ctx context.Context, obj client.Object) ([]graph.Vertex, error) {
+// HandleSwitchover is the implementation of the type Component interface method, which is used to handle the switchover of the Consensus workload.
+func (r *ConsensusSet) HandleSwitchover(ctx context.Context, obj client.Object) ([]graph.Vertex, error) {
```

Comment:
move `HandleSwitchover` implementation to StatefulSet base?

Reply:
I would expect this handling is common regardless of component type.

Reply:
+1 as duplicated in both consensus and replication

Reply:
> move `HandleSwitchover` implementation to StatefulSet base?

At present, stateless and stateful have not introduced the concept of roles (can be considered to support), and switchover depends on roles to switch, so here we first use component common util to eliminate redundant code, and subsequent new implementations can consider all workload unification .